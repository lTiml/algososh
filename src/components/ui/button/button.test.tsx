import { render, fireEvent } from "@testing-library/react";
import { Button, ButtonProps } from "./button";

describe("Button component", () => {
	const defaultProps: ButtonProps = {
		text: "",
		type: "button",
	}
	it('renders button with text correctly', () => {
		const wrapper = render(<Button {...defaultProps} text="Click me" />);
		expect(wrapper.container.firstChild).toMatchSnapshot();
	})
	it('renders button without text correctly', () => {
		const { container } = render(<Button {...defaultProps} text={undefined} />);
		expect(container.firstChild).toMatchSnapshot();
	})
	it('renders disabled button correctly', () => {
		const component = render(<Button {...defaultProps} disabled={true} />);
		expect(component.container.firstChild).toMatchSnapshot();
	})
	it('renders loading button correctly', () => {
		const component = render(<Button {...defaultProps} isLoader={true} />);
		expect(component.container.firstChild).toMatchSnapshot();
	})
	it('calls onClick callback when button is clicked', () => {
		const onClick = jest.fn();
		const button = render(<Button {...defaultProps} onClick={onClick} text="Click me" />);
		fireEvent.click(button.getByText("Click me"));
		expect(onClick).toHaveBeenCalledTimes(1);
	})
	it('renders button with text correctly', () => {
		const wrapper = render(<Button {...defaultProps} text="Click me" />);
		expect(wrapper.getByText("Click me")).toBeInTheDocument();
	})
	it('renders button without text correctly', () => {
		const { container } = render(<Button {...defaultProps} text={undefined} />);
		expect(container.querySelector('p')).toBeNull();
	})
	it('renders disabled button correctly', () => {
		const component = render(<Button {...defaultProps} disabled={true} />);
		expect(component.getByRole('button')).toBeDisabled();
	})
	it('renders loading button correctly', () => {
		const component = render(<Button {...defaultProps} isLoader={true} />);
		expect(component.getByAltText("Загрузка.")).toBeInTheDocument();
	})

})