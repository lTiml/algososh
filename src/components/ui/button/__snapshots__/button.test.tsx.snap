// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders button with text correctly 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`Button component renders button without text correctly 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
/>
`;

exports[`Button component renders disabled button correctly 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  
</button>
`;

exports[`Button component renders loading button correctly 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;
